includes/class-category-pager.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Newspack category pager fix.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack category pager.
+ *
+ * This class adds tweaks to sites where the permalink structure is "/%category%/%postname%/".
+ *
+ * See https://core.trac.wordpress.org/ticket/8905 and https://core.trac.wordpress.org/ticket/21209
+ */
+final class Category_Pager {
+	/**
+	 * Initialize Hooks.
+	 */
+	public static function init() {
+		add_action( 'init', [ __CLASS__, 'on_init' ] );
+	}
+
+	/**
+	 * Action callback for init.
+	 */
+	public static function on_init() {
```

Comment:
WDYT of hooking directly into `paginate_links` and checking the option there?

I think it's better to have that check done only when really needed than on every single request.

Reply:
Makes much better sense. I changed it. Thanks